rfc/rfc-2/2 - Deletion Info Conversion.md

```diff
@@ -0,0 +1,159 @@
+first locate the deletion vector, which can be stored either inline within the commit logs or can be stored in a 
+separate deletion vector file whose path is recorded in the commit logs. Once located, XTable can utilize the utility 
+methods provided by table format libraries to stream the ordinals of the records affected by the deletion operations.
+
+Unlike currently supported conversion of metadata in XTable, deletion vectors require special handling. While all
+table metadata is present in commit logs and manifest files, deletion vectors can be stored in separate files. The size
+of the deletion vectors is proportional to the number of records deleted, which can be significant in large tables.
+The proposed implementation aims to handle this efficiently by streaming the records from the sources to the targets.
+
+**New Class: `InternalDeletionVector`**
+To improve the functionality, a new class to manage the internal in-memory representation of deletion vectors is added.
+This is similar to creating internal representation for data files, schema, and partitioning spec.
+
+The following metadata is relevant for deletion vectors:
+- `dataFilePath`: Path of the data file associated with the deletion vector.
+- `size`: Size of the deletion vector.
+- `countRecordsDeleted`: Number of records deleted.
+- `sourceDeletionVectorFilePath`: Path of the deletion vector file (optional as deletion vectors can be stored inline).
+- `offset`: Offset of the deletion vector within the file (optional as inline deletion vectors do not have an offset).
+- `binaryRepresentation`: Binary representation of the inline deletion vector (optional).
+- `ordinalStream`: Stream of ordinals of the records deleted.
+
+**Modifications in `TableChange`**
+Currently XTable only captures the addition and removal of data files. To support deletion vectors, the `TableChange`
+class is updated to include the following fields a new field `deletionVectorsAdded` to capture deletion vectors added 
+in a commit. The sources will need to parse the commit logs to extract the deletion vectors and populate this field.
+
+**Updates in `DeltaActionsConverter`**
+The converter class is responsible for converting Delta Lake `actions` into internal representations, resulting in
+`TableChange` objects. The converter is updated to extract deletion vectors from the `AddFile` actions. Currently,
+the converters read content from source metadata files and then close the file handles. However, this will be different
+for deletion vectors. To avoid the potential memory explosion, the converters will create a provider of deletion vectors
+ordinals, which will be used to stream the ordinals of the records deleted to the targets.
+
+**Changes in `DeltaConversionSource`**
+As described above, the delta commit logs adds data file entries in commit log each time delete vectors are updated. 
+This needs special handling in the delta source, to avoid duplication of data file actions.
+
```

Comment:
I think it would be good to add a small section on the expectations for how the targets will handle the deletion vectors at a high level

Reply:
Added the high level details for Iceberg target. An integration test with iceberg changes is available here: [link](https://github.com/apache/incubator-xtable/commit/eb6a8c7b363d46d33bb82aa932aea18d12fe8f89#diff-b700eabfdce8dd7c376e280789e6be74d550c5b4a9be76de45185c7c73f94e9d)